doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,267 @@
+
+   See `this section`_ in CLion web help for more information on project
+   security.
+
+   .. _this section: https://www.jetbrains.com/help/clion/project-security.html#projects_security
+    </chapter>
+
+
+Create a CMake presets file
+***************************
+
+This presets_ file will activate the Python virtual environment for your Zephyr
+installation and will define the target board, the build directory,
+and the flashing tool.
+
+.. _presets: https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html
+
+You can find more information on how CLion works with CMake presets on
+`this web help page`_.
+
+.. _this web help page: https://www.jetbrains.com/help/clion/cmake-presets.html
+
+
+#. In the **Project** view, locate the ``my-workspace\example-application\app``
+   folder.
+
+#. Right-click the ``app`` folder and select **New | File**.
+
+#. Name the file ``CMakeUserPresets.json``.
+
+#. Add the following code into it:
+
+   .. code-block::
+
+      {
+         "version": 3,
+            "cmakeMinimumRequired": {
+            "major": 3,
+            "minor": 24
+         },
+         "configurePresets": [
+            {
+               "name": "Zephyr-Debug",
+               "generator": "Ninja",
+               "binaryDir": "build",
+               "cacheVariables": {
+                  "BOARD": "nucleo_f302r8",
+                  "CMAKE_PROGRAM_PATH": "&lt;OPENOCD_INSTALLATION_PATH&gt;;&lt;HOME_PATH&gt;/zephyrproject/.venv/Scripts"
+               }
+            }
+         ],
+         "buildPresets": [
+            {
+               "name": "Zephyr-Debug",
+               "configurePreset": "Zephyr-Debug"
+            }
+         ]
+      }
+
+#. Customize the following lines:
+
+   * ``"BOARD":...`` - specify the board you are using.
+
+   * ``"CMAKE_PROGRAM_PATH":...`` - set the path to Zephyr environment scripts
+     (delete this part if you are not using a virtual environment) and set
+     the path to your flashing software (OpenOCD in our example).
+
+#. Save the file. Make sure it is located in the ``app`` folder.
+
+
+Load CMake
+**********
+
+#. In the **Project** tree, locate the ``CMakeLists.txt`` file in the ``app``
+   folder.
+
+#. Right-click it and select **Load CMake Project**.
+
+   CMake load should finish successfully.
+
+
+Configure CMake profiles
+************************
+
+#. Go to **Settings | Build, Execution, Deployment | CMake**.
+
+#. Delete the automatically created **Debug** profile.
+
+#. Select the **Zephyr-Debug preset** profile and set the **Enable profile**
+   checkbox.
+
+#. Click **Apply** and close the dialog.
+
+   CMake project will be reloaded.
+
+
+Build the project
+*****************
+
+#. Check the top toolbar: the **Zephyr-Debug** profile and the
+   **zephyr_final** configuration should be selected.
+
+#. Click the hammer icon next to the configuration switcher.
+
+   When the build is finished, you can find build artifacts in
+   the ``app\build`` folder.
+
```

Comment:
Would be worth highlighting that other CMake targets can now of course be called, beyond just build (ex. `puncover`, `hardenconfig`, etc.)

Reply:
Added a sentence on that